src/main/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -760,6 +761,17 @@ private static boolean isVCFStream(final InputStream stream, final String MAGIC_
         return values;
     }
 
+    /**
+     * Parses a String as a Double, being tolerant for lower-case NaN (nan).
+     */
+    private static final Double decodeDouble(final String string) {
+        if (Pattern.matches("[+-]?nan", string)) {
```

Comment:
I don't see -+ nans defined anywhere, only "nan" without a sign...have you seen that in the wild? -+Infinity I have seen...but not NaN.

Could you please:
1. first attempt to call `Double.valueOf(string)` and  catch the exception if it fails (for performance)
2. `compile` the pattern into a static final (also for performance)
3.  make the pattern match case-independent `Pattern.compile(regex, Pattern.CASE_INSENSITIVE)` and remove the `[+-]` part.
4. also add checks (and tests) for `[-+]?inf(inity)?` and encode into either Double.POSITIVE_INFINITY or Double.NEGATIVE_INFINITY

Thanks!

Reply:
NaNs are NaNs and their sign is surely meaningless, but nonetheless the IEEE NaN format has a sign bit and C `printf` will print it (in glibc and other implementations, but not the BSD/macOS one — though IMHO that's not standard compliant). And Java will [parse it](https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html#valueOf-java.lang.String-).

So IMHO you should keep the `[+-]?` part. My suggestion for what the VCF spec should allow for NaN/∞ is samtools/hts-specs#409.

Reply:
I'd like to object to allowing `∞`...

Reply:
Some additional comments:

1.  Replying to @yfarjoun point 1:  Usually I'm not a fan of catching exceptions as part of the standard flow of a method, but in this case I agree that  it makes sense for performance reasons.
2.  This should be a replacement for `Double.parseDouble` and return a primitive `double` instead of the boxed `Double`